opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -870,21 +870,13 @@ namespace Opm {
             std::vector< Scalar > B_avg(numComponents(), Scalar() );
             computeAverageFormationFactor(B_avg);
 
-            if (param_.solve_welleq_initially_ && iterationIdx == 0) {
-                // solve the well equations as a pre-processing step
-                last_report_ = solveWellEq(B_avg, dt, local_deferredLogger);
-
-
-                if (initial_step_) {
-                    // update the explicit quantities to get the initial fluid distribution in the well correct.
-                    calculateExplicitQuantities(local_deferredLogger);
-                    prepareTimeStep(local_deferredLogger);
-                    last_report_ = solveWellEq(B_avg, dt, local_deferredLogger);
-                    initial_step_ = false;
-                }
-                // TODO: should we update the explicit related here again, or even prepareTimeStep().
-                // basically, this is a more updated state from the solveWellEq based on fixed
-                // reservoir state, will tihs be a better place to inialize the explict information?
+            if (param_.solve_welleq_initially_ && iterationIdx == 0  && initial_step_) {
+                // update the explicit quantities to get the initial fluid distribution in the well correct.
+                // the assembleWellEq will iterate the well solution until convergece.
+                assembleWellEq(B_avg, dt, local_deferredLogger);
```

Comment:
Consider adding `updateWellControls()` here. However, if controls are changed we might need to iterate, which is part of the reason for the loop in `solveWellEq()`... Doing it by individual well model would be an improvement in any case though.